arch/arm/core/cortex_m/mpu/arm_mpu_v8_internal.h

```diff
@@ -57,7 +57,7 @@ static void _region_init(u32_t index, struct arm_mpu_region *region_conf)
 }
 
 #if defined(CONFIG_USERSPACE) || defined(CONFIG_MPU_STACK_GUARD) || \
-	defined(CONFIG_APPLICATION_MEMORY)
+	defined(CONFIG_APPLICATION_MEMORY) || defined(CONFIG_RAM_FUNCTION)
```

Comment:
So, if CONFIG_RAM_FUNCTION is set, all the code below is compiled in, although we might only want _get_mpu_ramfunc_region_attr().